include/zephyr/sys/util.h

```diff
@@ -129,6 +129,21 @@ extern "C" {
 
 #endif /* __cplusplus */
 
+/**
+ * @brief Declare a flexible array member.
+ *
+ * This macro declares a flexible array member in a struct. The member
+ * is named @p name and has type @p type.
+ *
+ * The macro is specially useful for cases where flexible arrays are
+ * used in unions or are not the last element in the struct.
+ */
+#define FLEXIBLE_ARRAY_DECLARE(type, name) \
+	struct { \
+		struct { } __unused_##name; \
+		type name[]; \
+	}
+
```

Comment:
Given that this addition is for security, can we add any sort of unittesting for it? 

It probably should also be mentioned that flexible arrays shall still exist as the last field in a struct

Reply:
Yeah, not sure how to test such a macro. The only thing it can really test is that the variable name and type is proper

Reply:
would it make sense to have a note explaining why this should be used instead of the array directly? it's in the commit message but it seems like a good idea to have it in the macro documentation itself

Reply:
that is a good point. I will add it to the macro documentation but maybe it should also be added to the coding guidelines.